R/model.R

```diff
@@ -424,42 +424,82 @@ sample_method <- function(data = NULL,
-                          save_extra_diagnostics = FALSE,
+                          save_latent_dynamics = FALSE,
                           output_dir = NULL,
-                          num_chains = 4,
-                          num_cores = getOption("mc.cores", 1),
-                          num_warmup = NULL,
-                          num_samples = NULL,
+                          chains = 4,
+                          cores = getOption("mc.cores", 1),
+                          iter_warmup = NULL,
+                          iter_sampling = NULL,
                           save_warmup = FALSE,
                           thin = NULL,
-                          max_depth = NULL,
+                          max_treedepth = NULL,
                           adapt_engaged = TRUE,
                           adapt_delta = NULL,
-                          stepsize = NULL,
+                          step_size = NULL,
                           metric = NULL,
                           metric_file = NULL,
                           inv_metric = NULL,
                           init_buffer = NULL,
                           term_buffer = NULL,
                           window = NULL,
                           fixed_param = FALSE,
-                          validate_csv = TRUE) {
+                          validate_csv = TRUE,
+                          # deprecated
+                          num_cores = NULL,
+                          num_chains = NULL,
+                          num_warmup = NULL,
+                          num_samples = NULL,
+                          save_extra_diagnostics = NULL,
+                          max_depth = NULL,
+                          stepsize = NULL) {
 
-  checkmate::assert_integerish(num_chains, lower = 1, len = 1)
   if (fixed_param) {
-    num_chains <- 1
-    num_cores <- 1
+    chains <- 1
+    cores <- 1
     save_warmup <- FALSE
   }
+
+  # temporary deprecation warnings
+  if (!is.null(num_cores)) {
+    warning("'num_cores' is deprecated. Please use 'cores' instead.")
+    cores <- num_cores
+  }
+  if (!is.null(num_chains)) {
+    warning("'num_chains' is deprecated. Please use 'chains' instead.")
+    chains <- num_chains
+  }
+  if (!is.null(num_warmup)) {
+    warning("'num_warmup' is deprecated. Please use 'iter_warmup' instead.")
+    iter_warmup <- num_warmup
+  }
+  if (!is.null(num_samples)) {
+    warning("'num_samples' is deprecated. Please use 'iter_sampling' instead.")
+    iter_sampling <- num_samples
+  }
+  if (!is.null(max_depth)) {
+    warning("'max_depth' is deprecated. Please use 'max_treedepth' instead.")
+    max_treedepth <- max_depth
+  }
+  if (!is.null(stepsize)) {
+    warning("'stepsize' is deprecated. Please use 'step_size' instead.")
+    step_size <- stepsize
+  }
+  if (!is.null(save_extra_diagnostics)) {
+    warning("'save_extra_diagnostics' is deprecated. Please use 'save_latent_dynamics' instead.")
+    save_latent_dynamics <- save_extra_diagnostics
+  }
```

Comment:
I added these deprecation warnings to help current CmdStanR users, but these aren't strictly necessary since we haven't done a release yet. Thoughts on keeping vs removing these?

Reply:
I like them. Could we have one for threads = TRUE?

Reply:
I like them too, though I would remove them once we do v1.0.0? 

The threads one is more tricky and needs to be added to the compile step. I can take care of that one.

Reply:
> I like them too, though I would remove them once we do v1.0.0?

Yeah. I think we could even remove them for the beta release, but definitely for v1.0.0.